.github/workflows/label_good_first_prs.yml

```diff
@@ -0,0 +1,84 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2025 The Stdlib Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#/
+
+# Workflow name:
+name: label_good_first_prs
+
+# Workflow triggers:
+on:
+  pull_request_target:
+    types:
+      - opened
+      - closed
+      - synchronize
+      - reopened
+      - edited
+
+# Workflow jobs:
+jobs:
+
+  # Define a job which automatically labels pull requests based on whether they reference good first issues
+  labeler:
+
+    # Define job name:
+    name: 'Label PRs for issues with label "Good First Issue" as "Good First PR"s'
+
+    # Only run this job if the pull request did not have label `automated-pr`:
+    if: contains(github.event.pull_request.labels.*.name, 'automated-pr') == false
+
+    # Define job permissions:
+    permissions:
+      contents: read
+      pull-requests: write
+
+    # Define the type of virtual host machine:
+    runs-on: ubuntu-latest
+
+    # Define the sequence of job steps:
+    steps:
+      # Check whether any of the referenced issues is a "Good First Issue":
+      - name: 'Check whether any of the referenced issues is a "Good First Issue"'
+        id: 'check-pr'
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+        run: |
+          bool=$(. "$GITHUB_WORKSPACE/.github/workflows/scripts/references_good_first_issue $PR_NUMBER)
```

Comment:
@Planeshifter Do we need to pass along a GitHub token environment variable here?

Reply:
@kgryte I don't think so;`GITHUB_TOKEN` environment variable should be set by default.

In 

https://github.com/stdlib-js/stdlib/blob/3e3a5ead2a7e05e237dc0c5196949b77db17b11f/.github/workflows/check_contributing_guidelines_acceptance.yml#L88-L94

we need to set a different `GITHUB_TOKEN` since the default permissions are not sufficient, but here we are just reading the pull requests and issues.